client/state/comments/test/actions.js

```diff
@@ -0,0 +1,358 @@
+				comments: {
+					earliestCommentDate: Immutable.Map(),
+					queries: quiresMap
+				}
+			} );
+
+			requestPostComments( SITE_ID, POST_ID )( dispatchSpy, getStateSpy );
+
+			expect( dispatchSpy ).to.not.have.been.called;
+
+		} );
+
+
+		it('should dispatch correct first request actions', function( done ) {
+			this.timeout( 5000 );
+			const dispatchSpy = sinon.spy();
+			const getStateSpy = sinon.stub().returns( {
+				comments: {
+					earliestCommentDate: Immutable.Map(),
+					queries: Immutable.Map()
+				}
+			} );
+
+			const reqPromise = requestPostComments( SITE_ID, POST_ID )( dispatchSpy, getStateSpy );
+
+			expect( dispatchSpy ).to.have.been.calledWith( {
+				type: COMMENTS_REQUEST,
+				requestId: createRequestId( SITE_ID, POST_ID, { order: 'DESC', number: 50 } )
+			} );
+
+			reqPromise.then( () => {
+
+				expect( dispatchSpy ).to.have.been.calledWith( {
+					type: COMMENTS_REQUEST_SUCCESS,
+					requestId: createRequestId( SITE_ID, POST_ID, { order: 'DESC', number: 50 } )
+				} );
+
+				done();
+			}).catch( done );
+
+		} );
+
+		it('should dispatch correct consecutive request actions', function( done ) {
+			this.timeout( 5000 );
+
+			const beforeDateString = "2016-02-03T04:19:26.352Z";
+			const dispatchSpy = sinon.spy();
+			const getStateSpy = sinon.stub().returns( {
+				comments: {
+					earliestCommentDate: Immutable.fromJS( { [ createCommentTargetId( SITE_ID, POST_ID ) ]: new Date( beforeDateString ) } ),
+					queries: Immutable.Map()
+				}
+			} );
+
+			const reqPromise = requestPostComments( SITE_ID, POST_ID )( dispatchSpy, getStateSpy );
+
+			expect( dispatchSpy ).to.have.been.calledWith( {
+				type: COMMENTS_REQUEST,
+				requestId: createRequestId( SITE_ID, POST_ID, { order: 'DESC', number: 50, before: new Date( beforeDateString ) } )
+			} );
+
+			reqPromise.then( () => {
+				expect( dispatchSpy ).to.have.been.calledWith( {
+					type: COMMENTS_REQUEST_SUCCESS,
+					requestId: createRequestId( SITE_ID, POST_ID, { order: 'DESC', number: 50, before: new Date( beforeDateString ) } )
+				} );
+
+				done();
+			}).catch( done );
+
+		} );
+
+	}); // requestPostComments
+
+
+	describe( '#fetchAllCommentIds()', () => {
+		before( () => {
+			//nock.recorder.rec( {
+			//	output_objects: true
+			//} );
+			//nock( 'https://public-api.wordpress.com:443' ).record();
+		} );
+
+		after( () => {
+			//var nockCalls = nock.recorder.play();
+			//require('fs').writeFileSync( '/home/yury/1.txt', JSON.stringify( nockCalls ) );
+			//nock.restore();
+		} );
+
+		it( 'should fetch batched comments when exceeds MAX_NUMBER_OF_COMMENTS_IN_FETCH_RESULT', function ( done ) {
+			this.timeout( 3000 );
+
+			fetchAllCommentIds( SITE_ID, POST_ID ).then( (res) => {
+				expect( res.length ).to.equal( 122 );
+				done();
+			} ).catch( done );
+		} )
+	}); // fetchAllCommentIds
+
+
+	describe( '#writeComment()', () => {
+		before( () => {
+			nock('https://public-api.wordpress.com:443')
```

Comment:
@nb I think that mocking http requests has the additional benefit of testing my `actions` compliant with `wpcom` api if it might change in the future.

I've refactored the tests a bit, removed fixtures entirely and the [mock data](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-7683b0e3a8968ec7d9b075cd827dc5daR202) required now is relatively small, and the mock [itself is simple](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-7683b0e3a8968ec7d9b075cd827dc5daR225).